cmd/clusterctl/test/e2e/config_cluster_test.go

```diff
@@ -0,0 +1,130 @@
+// +build e2e
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"os"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	clusterctlclient "sigs.k8s.io/cluster-api/cmd/clusterctl/pkg/client"
+	clusterctlrepo "sigs.k8s.io/cluster-api/cmd/clusterctl/pkg/client/repository"
+	"sigs.k8s.io/cluster-api/test/framework/management/kind"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("clusterctl config cluster", func() {
+	Describe("basic", func() {
+		var (
+			cfgFile                string // default is $HOME/clusterctl.yaml
+			kindCluster            *kind.Cluster
+			kindClient             client.Client
+			clusterName            string
+			kubernetesVersion      string
+			coreProvider           string
+			coreVersion            string
+			bootstrapProvider      string
+			bootstrapVersion       string
+			infrastructureProvider string
+			infrastructureVersion  string
+			tmpDir                 string
+			template               clusterctlrepo.Template
+		)
+		BeforeEach(func() {
+			clusterName = "e2e-workload-cluster"
+			kubernetesVersion = "1.14.2"
+			coreProvider = "cluster-api"
+			coreVersion = "v0.3.0"
+			bootstrapProvider = "kubeadm-bootstrap"
+			bootstrapVersion = "v0.3.0"
+			infrastructureProvider = "docker"
+			infrastructureVersion = "v0.3.0"
+			var err error
+
+			// Create the mgmt cluster and client
+			kindCluster, kindClient, err = CreateKindClusterAndClient()
+			Expect(err).ToNot(HaveOccurred())
+
+			// Create clusterctl.yaml
+			tmpDir = createTempDir()
+			cfgFile = createLocalTestClusterCtlConfig(tmpDir, "clusterctl.yaml", "DOCKER_SERVICE_DOMAIN: \"docker.cluster.local\"")
+			// Let's setup some varibles for the workload cluster template
+			os.Setenv("DOCKER_SERVICE_CIDRS", "\"10.96.0.0/12\"")
+			os.Setenv("DOCKER_POD_CIDRS", "\"192.168.0.0/16\"")
+
+			c, err := clusterctlclient.New(cfgFile)
+			Expect(err).ToNot(HaveOccurred())
+			initOpt := clusterctlclient.InitOptions{
+				Kubeconfig:              kindCluster.KubeconfigPath,
+				CoreProvider:            coreProvider + ":" + coreVersion,
+				BootstrapProviders:      []string{bootstrapProvider + ":" + bootstrapVersion},
+				InfrastructureProviders: []string{infrastructureProvider + ":" + infrastructureVersion},
```

Comment:
Regarding the point made above, it would just be easier to provide
```
CoreProvider: coreProvider // cluster-api:v0.3.0
BootstrapProviders: []string{boostrapProvider} // kubeadm-bootstrap:v0.3.0
...
```